### PR TITLE
fix: rapid regrex does not fit extended version number

### PR DESCRIPTION
**Pull request by wchengru**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/local/docker/manager.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def _is_rapid_image(image_name: str) -> bool:
         : return bool: True, if the image name ends with rapid-$SAM_CLI_VERSION. False, otherwise
         """
 
-        if not re.search(r":rapid-\d+\.\d+.\d+$", image_name):
+        if not re.search(r":rapid-\d+\.\d+\.[\w\.]+$", image_name):
```
Comment (hawflau) [inline samcli/local/docker/manager.py:200]: nit: Can we add some examples of valid rapid image tags in the docstring?

Comment (wchengru) [reply samcli/local/docker/manager.py:202]: Done.

**Review by hawflau — APPROVED** [review]

**Review by torresxb1 — APPROVED** [review]

File: samcli/local/docker/lambda_image.py [hunk 2]
```diff
@@ -328,6 +328,6 @@ def is_rapid_image(image_name: str) -> bool:
         : return bool: True, if the image name ends with rapid-$SAM_CLI_VERSION. False, otherwise
         """
 
-        if not re.search(fr":{RAPID_IMAGE_TAG_PREFIX}-\d+\.\d+.\d+$", image_name):
+        if not re.search(fr":{RAPID_IMAGE_TAG_PREFIX}-\d+\.\d+\.[\w\.]+$", image_name):
```
Comment (mgrandis) [inline samcli/local/docker/lambda_image.py:331]: Could we just check that it starts with `rapid-`?

Comment (mgrandis) [reply samcli/local/docker/lambda_image.py:331]: Proposal:
```
        try:
            return image_name.split(":")[1].startswith(f"{RAPID_IMAGE_TAG_PREFIX}-")
        except (IndexError, AttributeError):
            # split() returned 1 or less items or image_name is None
            return False
```

Also adding tests in `tests/unit/local/docker/test_lambda_image.py`:
```
    def test_is_rapid_image(self):
        self.assertFalse(LambdaImage.is_rapid_image(None))
        self.assertFalse(LambdaImage.is_rapid_image(""))
        self.assertFalse(LambdaImage.is_rapid_image("my_repo"))
        self.assertFalse(LambdaImage.is_rapid_image("my_repo:tag"))

        self.assertTrue(LambdaImage.is_rapid_image("my_repo:rapid-1.29beta"))
        self.assertTrue(
            LambdaImage.is_rapid_image(f"public.ecr.aws/sam/emulation-python3.6:rapid-1.29.0")
        )
```

**Review by mgrandis — APPROVED** [review]

**Pull request merged**